canvas_workflow_helpers/protocols/example_protocols/phq9_banner.py

```diff
@@ -0,0 +1,74 @@
+from canvas_workflow_kit import events
+from canvas_workflow_kit.protocol import (
+    ClinicalQualityMeasure,
+    ProtocolResult,
+    STATUS_DUE,
+    STATUS_SATISFIED,
+)
+from canvas_workflow_kit.constants import CHANGE_TYPE, AlertPlacement, AlertIntent
+from canvas_workflow_kit.intervention import BannerAlertIntervention
+from canvas_workflow_kit.value_set.value_set import ValueSet
+import arrow
+
+
+class QuestionnairePhq9(ValueSet):
+    VALUE_SET_NAME = "PHQ-9 Questionnaire"
+    LOINC = {"44249-1"}
+
+
+class PHQ9HistoryBannerAlert(ClinicalQualityMeasure):
+    """
+    A Banner alert that displays the scores of the patient's last 3 completed PHQ9 questionnaires.
+    """
+
+    class Meta:
+        title = "PHQ9 History Banner Alert"
+
+        version = "2023-v01"
+
+        description = "A Banner alert that displays the scores of the patient's last 3 completed PHQ9 questionnaires."
+
+        information = "https://link_to_protocol_information"
+
+        identifiers = ["PHQ9HistoryBanner"]
+
+        types = ["CQM"]
+
+        responds_to_event_types = [
+            events.HEALTH_MAINTENANCE,
+        ]
+
+        compute_on_change_types = [CHANGE_TYPE.INTERVIEW]
+
+        authors = ["Canvas Example Medical Association (CEMA)"]
+
+    def get_date_and_score(self, q):
+        date = arrow.get(q.get("noteTimestamp")).format("MM/DD/YYYY")
+        score = next((r.get("score") for r in q.get("results")), "")
+        return f"{date}: {int(score)}"
+
+    def get_last_three_phq9(self):
+        return self.patient.interviews.find(QuestionnairePhq9).filter(
+            status="AC", progressStatus="F"
```

Comment:
again probably not necessary to even do the filtering but its okay

Reply:
Also what if we do 
```suggestion
            noteTimestamp__gte=arrow.now().shift(years=-1)
```
this instead and then you don't even need to check for the most recent result in compute_results and they can just be not applicable if they dont have a result within a year?

Reply:
but what if they have one a few days ago, and then several more from over year ago? wouldn't we still want to show all 3?

Reply:
Guess it doesnt hurt. Product can ask us to tweak it if it doesnt work for the demo